app/controllers/post_action_users_controller.rb

```diff
@@ -34,25 +24,36 @@ def index
         .order("post_actions.created_at ASC")
         .limit(page_size)
 
+    post_actions =
+      DiscoursePluginRegistry.apply_modifier(:post_action_users_list, post_actions, post)
+
     if !guardian.can_see_post_actors?(post.topic, post_action_type_id)
-      raise Discourse::InvalidAccess unless current_user
+      raise Discourse::InvalidAccess if current_user.blank?
       post_actions = post_actions.where(user_id: current_user.id)
     end
 
     action_type = PostActionType.types.key(post_action_type_id)
     total_count = post["#{action_type}_count"].to_i
-
+    post_actions = post_actions.to_a
     data = {
       post_action_users:
         serialize_data(
-          post_actions.to_a,
+          post_actions,
           PostActionUserSerializer,
-          unknown_user_ids: unknown_user_ids,
+          unknown_user_ids: current_user_muting_or_ignoring_users(post_actions.map(&:user_id)),
         ),
     }
 
     data[:total_rows_post_action_users] = total_count if total_count > page_size
 
     render_json_dump(data)
   end
+
+  def current_user_muting_or_ignoring_users(user_ids)
+    return [] if current_user.blank?
+    UserCommScreener.new(
```

Comment:
First time encountering `UserCommScreener`, but after a cursory look this seems appropriate, and a nice tucking away of the details pertaining to this. I assume you know how to use it. 👌

Reply:
Yes, since I know the author of UserCommScreener :P : 

![well-of-course-i-know-him-hes-me-ben-kenobi](https://github.com/discourse/discourse/assets/920448/388c9605-324f-498f-91cb-98126992fa2e)